lib/src/main/kotlin/com/swmansion/starknet/account/StandardAccount.kt

```diff
@@ -34,19 +37,61 @@ class StandardAccount(
     )
 
     override fun sign(calls: List<Call>, params: ExecutionParams): InvokeFunctionPayload {
-        val calldata = callsToExecuteCalldata(calls, params.nonce)
+        val calldata = callsToExecuteCalldata(calls)
         val tx = TransactionFactory.makeInvokeTransaction(
             contractAddress = address,
             entryPointSelector = selectorFromName(EXECUTE_ENTRY_POINT_NAME),
             calldata = calldata,
             chainId = provider.chainId,
             maxFee = params.maxFee,
             nonce = params.nonce,
-            version = params.version,
+            version = version,
         )
 
         val signedTransaction = tx.copy(signature = signer.signTransaction(tx))
 
         return signedTransaction.toPayload()
     }
+
+    override fun execute(calls: List<Call>, maxFee: Felt): Request<InvokeFunctionResponse> {
+        return getNonce().compose { nonce ->
+            val signParams = ExecutionParams(nonce = nonce, maxFee = maxFee)
+            val payload = sign(calls, signParams)
+
+            return@compose provider.invokeFunction(payload)
+        }
+    }
+
+    override fun execute(calls: List<Call>): Request<InvokeFunctionResponse> {
+        return estimateFee(calls).compose { estimateFee ->
+            val maxFee = FeeUtils.estimatedFeeToMaxFee(estimateFee.overallFee)
+            execute(calls, maxFee)
+        }
+    }
+
+    override fun getNonce(): Request<Felt> {
+        return provider.getNonce(address)
+    }
+
+    override fun estimateFee(calls: List<Call>): Request<EstimateFeeResponse> {
+        return getNonce().compose { buildEstimateFeeRequest(calls, it) }
+    }
+
+    private fun buildEstimateFeeRequest(calls: List<Call>, nonce: Felt): Request<EstimateFeeResponse> {
+        val executionParams = ExecutionParams(nonce = nonce, maxFee = Felt.ZERO)
```

Comment:
We're trying to build an estimate fee request for the transaction and at this point it is unknown. Zero seems like a reasonable default value since it shouldn't affect the estimation anyways.